kayttooikeus-service/src/main/java/fi/vm/sade/kayttooikeus/service/impl/KutsuServiceImpl.java

```diff
@@ -251,8 +250,56 @@ public String createHenkilo(String temporaryToken, HenkiloCreateByKutsuDto henki
         kutsuByToken.setLuotuHenkiloOid(henkiloOid);
         kutsuByToken.setTila(KutsunTila.KAYTETTY);
 
+
+        // Set henkilo to VIRKAILIJA since we don't know if he was OPPIJA before
+        HenkiloUpdateDto henkiloUpdateDto = new HenkiloUpdateDto();
+        henkiloUpdateDto.setOidHenkilo(henkiloOid);
+        henkiloUpdateDto.setHenkiloTyyppi(HenkiloTyyppi.VIRKAILIJA);
+
+        // In case henkilo already exists
+        henkiloUpdateDto.setKutsumanimi(henkiloCreateDto.getKutsumanimi());
+
+        if(isNewHenkilo) {
+            henkiloUpdateDto = addEmailToNewHenkiloUpdateDto(henkiloUpdateDto, kutsuByToken.getSahkoposti());
+        } else {
+            henkiloUpdateDto = addEmailToExistingHenkiloUpdateDto(henkiloOid, kutsuByToken.getSahkoposti(), henkiloUpdateDto);
+        }
+
         this.ldapSynchronizationService.updateHenkiloAsap(henkiloOid);
-        return henkiloOid;
+        return henkiloUpdateDto;
+    }
+
+    private HenkiloUpdateDto addEmailToExistingHenkiloUpdateDto(String henkiloOid, String kutsuSahkoposti, HenkiloUpdateDto henkiloUpdateDto) {
+        HenkiloDto henkiloDto = this.oppijanumerorekisteriClient.getHenkiloByOid(henkiloOid);
+
+        // Initiate new YhteystiedotRyhma with email in kutsu
+        YhteystietoDto yhteystietoDto = new YhteystietoDto(YhteystietoTyyppi.YHTEYSTIETO_SAHKOPOSTI, kutsuSahkoposti);
+        HashSet<YhteystietoDto> yhteystietoDtos = new HashSet<>();
+        yhteystietoDtos.add(yhteystietoDto);
+        Set<YhteystiedotRyhmaDto> yhteystiedotRyhma = new HashSet<>();
+
+        // collect existing henkilos all yhteystiedot to a list
+        HashSet<YhteystietoDto> existingYhteystiedot = new HashSet<>();
```

Comment:
Nämä 3 riviä saa yhteen streamiin flatmapilla. Jos näitä ei käytetä missään muualla niin voisi suoraan päättää streamin .nonematch() ja ottaa booleanin talteen.